src/modules/registrypreview/RegistryPreviewUILib/RegistryPreviewMainPage.DataPreview.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Foundation.Metadata;
+
+namespace RegistryPreviewUILib
+{
+    public sealed partial class RegistryPreviewMainPage : Page
+    {
+        internal void ShowEnhancedDataPreview(string name, string type, string value)
+        {
+            // Create dialog
+            var panel = new StackPanel()
+            {
+                Spacing = 16,
+            };
+            ContentDialog contentDialog = new ContentDialog()
+            {
+                // Title = "Value preview",
+                Title = "View data - " + name,
+                Content = panel,
+                CloseButtonText = "Close",
+                DefaultButton = ContentDialogButton.None,
+            };
+
+            // Add content based on value type
+            switch (type)
+            {
+                case "REG_DWORD":
+                case "REG_QWORD":
+                    var hexBox = new TextBox()
+                    {
+                        Header = "Hexadecimal",
+                        IsReadOnly = true,
+                        Text = value.Split(" ")[0],
+                    };
+                    var decimalBox = new TextBox()
+                    {
+                        Header = "Decimal",
+                        IsReadOnly = true,
+                        Text = value.Split(" ")[1].TrimStart('(').TrimEnd(')'),
+                    };
+                    panel.Children.Add(hexBox);
+                    panel.Children.Add(decimalBox);
+                    break;
+                case "REG_BINARY":
+                case "REG_MULTI_SZ":
+                    var multiLineBox = new TextBox()
+                    {
+                        IsReadOnly = false,
+                        Text = "line 1 \r line 2 \r line 3",
+                        AcceptsReturn = true,
+                        MaxHeight = 200,
+                        TextWrapping = TextWrapping.NoWrap,
```

Comment:
@niels9001
Why isn't the multi line text box working?
![image](https://github.com/user-attachments/assets/3cc95768-2ce2-433d-8f04-8db6f1408206)

Reply:
Note to me: Try to first init and then set text.

Reply:
This is solved. The property order in code was wrong.